README.md

```diff
@@ -17,38 +23,215 @@ npm i @dialonce/boot
 
 ```js
-const logger = require('@dialonce/boot')().logger;
-const notifier = require('@dialonce/boot')().notifier;
+const { chain, logger, notifier } = require('@dialonce/boot')();
 ```
 
 *Please note that these instructions will print an error if the module has not been initialised before*
 
-## Logger error/warn forward
-If you use the logger with error/warn level, the message is automatically
-forwarded to the notifier. You still have the ability to disable it globally or
-locally.
+A ``chain`` object is an instance of a ``LoggerChain`` class, which implements a Chain of Responsibility pattern, while ``logger`` and ``notifier`` are adapters to make the resulting interface more user friendly.
+
+**NOTE!** If you use ``chain.log()`` function, your message will travel along the chain and each chain link will process it.
+``logger`` is a class, used to provide the same interface as ``winston`` package does. Although, it still sends a message to the ``chain.log`` function internally.
+
+However, a ``notifier`` is a single node, which will process the message outside the chain.
+
+This decision was made out of the idea that ``notifier`` might be used in some situations when you just want an error to be sent to bugsnag omitting all the other chains.
+
+In a general case, you might want to use a ``logger`` or a ``chain`` objects.
+
+### Interface
+
+**Logger** exposes the following set of data:
+```js
+// logging functions
+logger.silly(message, ...metadata);
+logger.verbose(message, ...metadata);
+logger.debug(message, ...metadata);
+logger.info(message, ...metadata);
+logger.warn(message, ...metadata);
+logger.error(message, ...metadata);
+logger.log(logLevel, message, ...metadata);
+
+// profiling function
+logger.profile(label);
+
+// properties
+logger.loggerChain; // instance of a logger chain
+```
+
+**Notifier** exposes:
+```js
+// function
+notifier.notify(message, ...metadata);
+
+// property
+notifier.settings // configuration object
+notifier.requestHandler // standard bugsnag requestHandler middleware
+```
 
+**Chain** exposes:
 ```js
-const logger = require('@dialonce/boot')({ NOTIFY: false }).logger; // to disable it globally
+// functions
+chain.log(logLevel, message, ...metadata);
+
+//properties
+chain.settings; // configuration object
+chain.bugsnagChain; // Bugsnag chain link
+chain.logentriesChain; // Logentries chain link
+chain.consoleChain; // Console chain link
+chain.chainStart; // first chain link (console by default)
+chain.chainEnd; // last chain link (bugsnag by default)
+chain.ChainLink; // ChainLink class - used to create a custom chainLink
 ```
 
+### Internal implementation details
+Firstly, you call any of the either ``logger`` or ``notifier`` or ``chain`` function.
+
+Then provided data is then converted into a ``frozen`` (``Object.freeze()``) message package object:
```

Comment:
maybe leave out Object.freeze() and just say it is immutable (you also use deepfreeze now).